docs/developers/debugging.md

```diff
@@ -0,0 +1,133 @@
+
+Using environment variables for CLI arguments allows these values to be reused across IDEs and IDE configurations seamlessly.
+
+
+## P2P Network
+
+When running the client with the P2P service enabled, i.e., building the binary with the `p2p` feature and supplying the runtime argument `--enable-p2p`, the client will connect to a Fuel network. This requires additional CLI arguments, including the `--keypair` and `--network` arguments.
+
+A key pair can be generated by running the utility binary `fuel-core-keygen`:
+
+```bash
+cargo run --bin fuel-core-keygen new
+```
+
+The resulting key pair printed to the console contains an address and secret. The `secret` can be provided to the `--keypair` argument when running the node.
+
+The `--network` argument identifies the name of the network to join. The network name is used during peer discovery. For example, users can specify `--network beta-4` to join the Beta 4 network. Similarly, setting the environment variable `NETWORK="beta-4"` will produce the same result.
+
+For more information about client networking, see the Fuel guide on [running a node](https://docs-hub.vercel.app/guides/running-a-node/).
+
+## Recommended IDEs
+
+During development, it is recommended to delegate the execution of Cargo instructions to an IDE and connect the application to the IDE's debugger, instead of running these instructions manually.
+
+Recommended IDEs include:
+- CLion
+- Visual Studio Code
+
+## CLion
+
+### Prerequisites
+
+Running Fuel Core with CLion requires the official JetBrains Rust plugin. Find the Rust plugin by navigating to CLion's Plugin menu, entering "Rust" in the search. Install it by clicking "Install".
+
+### Create A Run Configuration
+
+1. In the top-right corner of the CLion window, click on the configurations dropdown menu and select "Edit Configurations..."
+2. In the "Run/Debug Configurations" window, click on the "+" button and select "Cargo"
+3. In the "Name" field, give your configuration a descriptive name, such as "Run Beta 3"
+4. In the "Command" field, enter `run --all-features --bin fuel-core -- run <ARGUMENTS>`, where `<ARGUMENTS>` represents the desired CLI arguments. Refer to the CLI documentation for guidance on which options and arguments to use.
+5. If you use environment variables for configuration (as indicated in the CLI documentation), you can set them in the "Environment variables" section. Enter the variables and their values as needed. It is recommended to set the `RUST_LOG` environment variable to either `info` or `debug`.
+6. Click "Apply" or "OK" to save your run configuration.
+
+### Example Configuration
+
+![clion_configuration.png](assets/clion_configuration.png)
```

Comment:
Following the steps, I had to do;
5. Select the workspace folder
6. Remove on the before start section the build action

Reply:
Once I try to run it throws the error: 
```
ERROR fuel_core::cli: 57: Unable to load .env environment variables: path not found. Please check that you have created a .env file in your working directory.
```

Reply:
Thanks for going through the steps!

I think it would be a good idea for me to add a quick line adding the workspace folder. Usually, CLion should populate this line for you to your repository root (don't change it to what's exactly in the example!) But if that is missing for whatever reason, it would be helpful to know that you should set this to the root yourself. I see that I omitted that instruction.

The "Before start" settings are optional. CLion adds "Build" by default, but it's okay because `cargo run` does a build by default. I will add a note on these settings too.

Reply:
> Once I try to run it throws the error:
> 
> ```
> ERROR fuel_core::cli: 57: Unable to load .env environment variables: path not found. Please check that you have created a .env file in your working directory.
> ```

This is actually a new feature and it's deliberate. It simply means the client is expecting you to provide a `.env` file (probably because you compiled with `--all-features`), but it didn't find it. Simply add a `.env` file in the root of your repo with your specific configuration values, and you're good to go.

Reply:
Yeah, it worked even with the error. The real error was on the database folder: 

```
thread 'main' panicked at 'source slice length (64) does not match destination slice length (32)', <>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/fuel-types-0.36.0/src/array_types.rs:389:16